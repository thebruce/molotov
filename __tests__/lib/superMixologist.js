const Molotov = require('../../lib/molotov');
const SuperMixologist = require('../../lib/superMixologist');
const { MolotovError } = require('../../lib/molotovError');
const _ = require('lodash');

const molotovConfig = {
  testMolotovImplementer: {
    supersNameSpace: {
      testSuper: 'testSuper'
    },
    molotovPlugins: {
      testSuper: {
        onlyPluginOne: [
          'pluginOne',
        ],
        pluginAll: [
          'pluginOne',
          'pluginTwo',
          'pluginThree',
        ],
        pluginOneTwo: [
          'pluginOne',
          'pluginTwo',
        ],
      },
    },
  },
};

const configOverrides = {
  testMolotovImplementer: {
    molotovPlugins: {
      testSuper: {
        pluginAll: [
          'pluginOne',
          'pluginTwo',
        ],
        pluginOneThree: [
          'pluginOne',
          'pluginThree',
        ],
      },
    },
  },
};

const mixins = require('./../helpers/mixins');
const supers = require('./../helpers/supers');
const cocktailClass = require('./../helpers/cocktailHelper');

const molotov = new Molotov(molotovConfig, 'testMolotovImplementer', supers, mixins, configOverrides, [cocktailClass]);

let tmpMocks = [];

describe('Super Mixologist basic', () => {
  beforeEach(() => {
    tmpMocks.forEach(mock => mock.mockRestore());
    tmpMocks = [];
    jest.resetAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  test('Super Mixologist Constructor tests', () => {
    expect.assertions(3);
    const superMixologist = new SuperMixologist(molotov);
    expect(superMixologist.getType()).toBe('Supers');
    expect(superMixologist.getTarget()).toBe('supersNameSpace');
    expect(superMixologist.molotov).toBeInstanceOf(Molotov);
  });
});

describe('Super Mixologist resolve', () => {
  beforeEach(() => {
    tmpMocks.forEach(mock => mock.mockRestore());
    tmpMocks = [];
    jest.resetAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  test('Super Mixologist fetchOverrides tests', () => {
    expect.assertions(1);
    const superMixologist = new SuperMixologist(_.cloneDeep(molotov));
    expect(superMixologist.fetchOverrides()).toMatchSnapshot();
  });
});

describe('Super Mixologist resolve', () => {
  beforeEach(() => {
    tmpMocks.forEach(mock => mock.mockRestore());
    tmpMocks = [];
    jest.resetAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  test('Super Mixologist resolve tests', () => {
    expect.assertions(1);
    const superMixologist = new SuperMixologist(_.cloneDeep(molotov));
    expect(superMixologist.resolve()).toMatchSnapshot();
  });
});