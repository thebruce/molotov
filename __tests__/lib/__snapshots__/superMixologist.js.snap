// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Super Mixologist resolve Super Mixologist fetchOverrides tests 1`] = `
Object {
  "testMolotovImplementer": Object {
    "molotovPlugins": Object {
      "testSuper": Object {
        "onlyPluginOne": Array [
          "pluginOne",
        ],
        "pluginAll": Array [
          "pluginOne",
          "pluginTwo",
          "pluginThree",
        ],
        "pluginOneTwo": Array [
          "pluginOne",
          "pluginTwo",
        ],
      },
    },
    "supersNameSpace": Object {
      "testSuper": "testSuper",
    },
  },
}
`;

exports[`Super Mixologist resolve Super Mixologist resolve tests 1`] = `
Object {
  "testSuper": [Function],
  "testSuperTwo": [Function],
}
`;
